src/contexts/interactiveComponentsContext.tsx

```diff
@@ -6,32 +6,55 @@ export interface InteractiveComponentsState {
   toggleSidebar: () => void;
   selectedIndex: string;
   selectIndex: (index: string) => void;
+  isProfileModalOpened: boolean;
+  closeProfileModal: () => void;
+  openProfileModal: () => void;
 }
 
 const InteractiveComponentsDefaultState: InteractiveComponentsState = {
   isSidebarOpened: true,
   toggleSidebar: () => null,
   selectedIndex: '',
   selectIndex: () => null,
+  isProfileModalOpened: false,
+  closeProfileModal: () => null,
+  openProfileModal: () => null,
 };
 
 export const InteractiveComponentsContext = createContext(InteractiveComponentsDefaultState);
 
 export const InteractiveComponentsProvider = ({ children }: any) => {
   const [isSidebarOpened, setIsSidebarOpened] = useState(true);
   const [selectedIndex, setSelectedIndex] = useState(SIDEBAR_BUTTONS_LIST[0].key);
+  const [isProfileModalOpened, setIsProfileModalOpened] = useState(false);
```

Comment:
Context should have only 1 responsibility: Send the state from hooks to children components of ContextProvider.

Here you are missing the hooks calling. You never should define the state in the Context of provider, you should call the hook which define the state and business logic of the component you are creating.

So I would say you have here to do next things:

1. Create hooks for the sidebar and generic modal component (as Silviu suggested)
2. Move all the states and business logic (including functions) inside the hook
3. Call the hook in the provider, extract the state and functions you need to pass further and that's all.

Reply:
Have updated the logic in the **interactiveComponentsContext**, only for the **modal** for now. If that is good to go, will create a separate ticket to address the changes for the whole **interactiveComponentsContext**.